db/flush_job.cc

```diff
@@ -360,9 +640,12 @@ Status FlushJob::WriteLevel0Table() {
                          << total_memory_usage << "flush_reason"
                          << GetFlushReasonString(cfd_->GetFlushReason());
 
-    {
+    // In certain situations, a non-empty mems_ can still lead
+    // to memtables.empty(). For example, if mems_ is made of
+    // memtables that have been mempurged.
+    if (!memtables.empty()) {
       ScopedArenaIterator iter(
-          NewMergingIterator(&cfd_->internal_comparator(), &memtables[0],
+          NewMergingIterator(&cfd_->internal_comparator(), memtables.data(),
```

Comment:
Use of vector.data() is preferable (allowed in C++11), because &memtables[0] leads to an undefined behavior if memtables is empty (I also added an if-statment, pure paranoia).